app/api/csv/specs/typeParsers.spec.js

```diff
@@ -110,4 +110,120 @@ describe('csvLoader typeParsers', () => {
       }
     );
   });
+
+  describe('multidate', () => {
+    it.each`
+      dateProp                                 | dateFormat      | expectedDate
+      ${'2014|2015'}                           | ${'dd/MM/yyyy'} | ${['01-01-2014', '01-01-2015']}
+      ${'2014 11 6|2015 11 6'}                 | ${'dd/MM/yyyy'} | ${['06-11-2014', '06-11-2015']}
+      ${'December 17, 1995|December 17, 1996'} | ${'dd/MM/yyyy'} | ${['17-12-1995', '17-12-1996']}
+      ${'12/01/2021|12/01/2022'}               | ${'dd/MM/yyyy'} | ${['12-01-2021', '12-01-2022']}
+      ${'2021/12/01|2022/12/01'}               | ${'yyyy/MM/dd'} | ${['01-12-2021', '01-12-2022']}
+    `(
+      "should parse '$dateProp' with format '$dateFormat' and return a timestamp",
+      async ({ dateProp, dateFormat, expectedDate }) => {
+        const templateProp = { name: 'date_prop' };
+
+        const expected = await typeParsers.multidate(
+          { date_prop: dateProp },
+          templateProp,
+          dateFormat
+        );
+
+        expect(expected.map(date => moment.utc(date.value, 'X').format('DD-MM-YYYY'))).toEqual(
+          expectedDate
+        );
+      }
+    );
+  });
+
+  describe('daterange', () => {
+    it.each`
+      dateProp                                 | dateFormat      | expectedDate
+      ${'2014:2015'}                           | ${'dd/MM/yyyy'} | ${{ from: '01-01-2014', to: '01-01-2015' }}
+      ${'2014 11 6:2015 11 6'}                 | ${'dd/MM/yyyy'} | ${{ from: '06-11-2014', to: '06-11-2015' }}
+      ${'December 17, 1995:December 17, 1996'} | ${'dd/MM/yyyy'} | ${{ from: '17-12-1995', to: '17-12-1996' }}
+      ${'12/01/2021:12/01/2022'}               | ${'dd/MM/yyyy'} | ${{ from: '12-01-2021', to: '12-01-2022' }}
+      ${'2021/12/01:2022/12/01'}               | ${'yyyy/MM/dd'} | ${{ from: '01-12-2021', to: '01-12-2022' }}
+    `(
+      "should parse '$dateProp' with format '$dateFormat' and return the to and from timestamps",
+      async ({ dateProp, dateFormat, expectedDate }) => {
+        const templateProp = { name: 'date_prop' };
+
+        const expected = await typeParsers.daterange(
+          { date_prop: dateProp },
+          templateProp,
+          dateFormat
+        );
+
+        expect({
+          from: moment.utc(expected[0].value.from, 'X').format('DD-MM-YYYY'),
+          to: moment.utc(expected[0].value.to, 'X').format('DD-MM-YYYY'),
+        }).toEqual(expectedDate);
+      }
+    );
+  });
+
+  describe('multidaterange', () => {
+    it.each([
+      {
+        dateProp: '2014:2015|2016:2017',
```

Comment:
If you think it is worth it, the same extra cases as in the multidate case.

Reply:
Same as multidate